source/security/index.md

```diff
@@ -289,3 +289,38 @@ If you were using `excludeParams` previously please compare patterns used by yo
+
+CSP is enabled by setting the `Content-Security-Policy` or `Content-Security-Policy-Report-Only` HTTP response headers, based on whether it’s in enforcing or reporting mode, respectively. In enforcing mode, any event that violates the policy is not executed and if the policy specifies  a `reportUri` a JSON object with the details of the CSP violation is sent to the specified endpoint. In reporting mode, the event is executed normally and a CSP report is being sent to the defined `reportUri`.
+
+CSP comes in many flavors, but the policy we chose to implement is a nonce-based strict-dynamic CSP, which is the most robust policy. This policy requires the browser to check that all scripts have the valid nonce value (which is included in the response header generated by the server) before running them. This ensures that a malicious script not approved (by attaching the nonce value) by the server will not run, thus protecting against XSS attacks. With each response, a new nonce value is generated, so adopting CSP requires implementing a mechanism that automatically attaches the valid nonce value to all `script, link` blocks. Strict CSP also blocks any inline event handlers (`onclick` etc.) and `javascript:` URIs, so adopting this policy requires refactoring by the developers if these were in use. The default recommended policy used is as follows:
+
+```
+Content-Security-Policy:
+  object-src 'none';
+  script-src 'nonce-{random}' 'strict-dynamic' https: http:;
+  base-uri 'none';
+  report-uri https://your-report-collector.example.com/csp-report
+```
+
+The specified policy includes a `report-uri` directive. This should not be confused with the report-only and enforcing modes of CSP. In both modes, if the `report-uri` is specified, the browser will send a JSON blob containing details about the violation as a POST request to the specified `report-uri`.
+
+```
+{ "csp-report": {
+"document-uri":"http://localhost:8080/maven-example/searchForm.action",
+"referrer":"http://localhost:8080/maven-example/search",
+"violated-directive":"script-src-elem",
+"effective-directive":"script-src-elem",
+"original-policy":"object-src 'none'; script-src 'nonce-ryxoOreH2A9wuoB-gQr7S6ta' 'strict-dynamic' http: https:; base-uri 'none'; report-uri /csp-reports",
+"disposition":"report","blocked-uri":"inline","line-number":8,"source-file":"http://localhost:8080/maven-example/searchForm.action",
+"status-code":200,
+"script-sample":""}
+}
+```
+
+If a report-uri is not specified developers can see the violation in the dev tools of their browser. Reporting is an important aspect of adopting CSP, because the strict-dynamic CSP we’re proposing may break parts of an application (like using inline event handlers etc.) and need refactoring. Not having access to accurate CSP reports to know what is causing a violation will make developers less likely to adopt CSP. An example CSP report of a violation for a script block without a valid nonce-value looks like the above block.
```

Comment:
```suggestion
Regardless of whether a report URI is specified, violation are always shown on DevTools. Reporting is an important aspect of adopting CSP because restrictive policies might require refactoring of existing dangerous code patterns (like  inline event handlers) before the policy can be set to enforcement mode. Developers are encouraged to set a report URI and examine violation reports before enforcing their policies.
```